datamol/utils/fs.py

```diff
@@ -327,3 +340,112 @@ def glob(path: str, **kwargs) -> List[str]:
         data_paths = [f"{protocol}://{d}" for d in data_paths]
 
     return data_paths
+
+
+def recursive_ls(dir_path: str, with_directory: bool = True, _fs=None):
```

Comment:
> I am getting: gcs:// as the protocol here. Running any file with that on gsutil ls xxx results in InvalidUrlError: Unrecognized scheme "gcs".

That's annoying... I am not sure how to fix that. Any idea?

> In general, I use the glob scheme: output = fs.glob("gs://path/of/interest/**") which works quite well and if you want dir vs files, fs.isdir(f) can be applied on the item of the output

As you said, this function is because of `fs.isdir(f)`. I was afraid it will trigger a remote call for every single files/dirs which could dramatically slow down the function. Unless, you've already tested it with `copy_dir`, I will let it as it is. `fs.glob` is still here anyway.

Reply:
> That's annoying... I am not sure how to fix that. Any idea?

str replace of the scheme

Otherwise, `glob` use `find` / `ls`  (see https://filesystem-spec.readthedocs.io/en/latest/_modules/fsspec/spec.html#AbstractFileSystem.glob) 

This particular line should solve the issue as it returns a dict with key   `'StorageClass'` that tells if  it's a `'DIRECTORY'` or `'STANDARD'` file.

`allpaths = self.find(root, maxdepth=depth, withdirs=True, detail=True, **kwargs)`

Reply:
I removed `recursive_ls` in favor of `self.find` (`glob` is a no-go, as it is too inconsistent across the different backends).

I tested `copy_dir` for local <-> gcs, local <-> local and gcs <-> gcs and they all work.